pkg/ctl/gitops/apply.go

```diff
@@ -0,0 +1,182 @@
+package gitops
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"github.com/weaveworks/eksctl/pkg/gitops/flux"
+	"github.com/weaveworks/eksctl/pkg/utils/file"
+)
+
+type options struct {
+	gitOptions           git.Options
+	quickstartNameArg    string
+	outputPath           string
+	gitPrivateSSHKeyPath string
+}
+
+func applyGitops(cmd *cmdutils.Cmd) {
+	cfg := api.NewClusterConfig()
+	cmd.ClusterConfig = cfg
+
+	cmd.SetDescription("apply", "Setting up GitOps and apply a Quick Start profile", "")
+
+	var opts options
+
+	cmd.SetRunFuncWithNameArg(func() error {
+		return doApplyGitops(cmd, opts)
+	})
+
+	cmd.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVarP(&opts.quickstartNameArg, "quickstart-profile", "", "", "name or URL of the Quick Start profile. For example, app-dev.")
+		fs.StringVarP(&opts.gitOptions.URL, "git-url", "", "", "URL for the git repository that will contain the cluster components")
+		fs.StringVarP(&opts.gitOptions.Branch, "git-branch", "", "master", "Git branch")
+		fs.StringVarP(&opts.outputPath, "output-path", "", "./", "Path to directory where the GitOps repo will be cloned")
+		fs.StringVar(&opts.gitOptions.User, "git-user", "Flux", "Username to use as Git committer")
+		fs.StringVar(&opts.gitOptions.Email, "git-email", "", "Email to use as Git committer")
+		fs.StringVar(&opts.gitPrivateSSHKeyPath, "git-private-ssh-key-path", "",
```

Comment:
Sorry I missed this. I think that if we add this flag here, it should also be part of the `eksctl install flux`

Reply:
And, the inconsistency between `eksctl gitops apply` and `eksctl install flux` is going to be confusing for the end users.

Reply:
`install flux' already has this. It was added by marc

Reply:
Ah ... sorry, I missed that